### DISCUSSION TITLE
Add imageDidLoad callback to Image props?

### DISCUSSION THREAD

**Discussion by NickTheFreak97** [chakra-ui/chakra-ui#8547] [Ideas]
### Description

I stumbled upon a few use cases where it would be a huge improvement in quality of life to be able to attach a callback to execute when the image finished loading. 

For instance if I try to wrap @chakra-ui/Image inside [react-zoom-pinch-pan](https://www.npmjs.com/package/react-zoom-pan-pinch) and the image is on a remote server, the image never appears (I suppose it's because the latter library copies the wrapped component on mount). An easy fix would be something like:

```
{ imageDidLoad ? 
    <TransformWrapper>
        <TransformComponent>
             <Image
                    objectFit="cover"
                    src={image}
                    alt={description}
                    width="100%"
                    height="100%"
                    imageDidLoad={ (sourceURL) => {
                        this.setImageDidLoad(true)
                    }}
               />
        </TransformComponent>
    </TransformWrapper>
        :
    <ImageSkeleton
         width="100%"
         height="100%"
     />
}
```

### Problem Statement/Justification

Reacting to image loading event improves compatibility of Chakra-ui with other external libraries as in the example above. Such a prop can also be used to display a Skeleton or a placeholder more complex than just a text before the image remote image is rendered.

### Proposed Solution or API

Add to `Image.props` the following

`imageDidLoad`
Description
The callback invoked when the image with a remote URL is specified in the `src` prop and the remote image is first rendered.

Type
(remoteURL: String) => void

### Alternatives

Currently there's no way to test whether or not a remote image did load.

**Comment by segunadebayo** [comment]
Use the native `onLoad` on `Image`. No extra callback.
